core/invoker/src/main/scala/org/apache/openwhisk/core/containerpool/kubernetes/KubernetesContainerFactory.scala

```diff
@@ -40,12 +41,25 @@ import org.apache.openwhisk.core.{ConfigKeys, WhiskConfig}
 class KubernetesContainerFactory(
   label: String,
   config: WhiskConfig,
+  containerArgsConfig: ContainerArgsConfig = loadConfigOrThrow[ContainerArgsConfig](ConfigKeys.containerArgs),
   runtimesRegistryConfig: RuntimesRegistryConfig = loadConfigOrThrow[RuntimesRegistryConfig](
     ConfigKeys.runtimesRegistry))(implicit actorSystem: ActorSystem, ec: ExecutionContext, logging: Logging)
     extends ContainerFactory {
 
   implicit val kubernetes = initializeKubeClient()
 
+  val augmentedEnv: Map[String, String] = {
+    val envArgs = containerArgsConfig.extraArgs.filter((t) => t._1.equals("env")).values
+    envArgs
+      .flatMap(
+        _.filter(_.contains("="))
+          .map(x => {
+            val tmp = x.split("=")
```

Comment:
You can make use of flatMap to fold this map and the filter into one operation I think:

```scala
val envArgs = ...getOrElse("env", Set.empty[String])
envArgs
  .flatMap(_.split("=") match {
    case Array(key, value) => Some(key -> value)
    case _ => None
  }.toMap
```

Reply:
@markusthoemmes good proposal - this one should work:

```
    val envVars: Set[String] = containerArgsConfig.extraArgs.getOrElse("env", Set.empty[String])
    val augmentedEnv: Map[String, String] = envVars.flatMap {
      _.split("=", 2) match {
        case Array(key, value) => Some(key -> value)
        case _                 => None
      }
    }.toMap
```